### ISSUE TITLE
Syslog format in 0.11.1 still has unwanted padding

### ISSUE THREAD

**Issue by hlein** [fail2ban/fail2ban#2734]
### Environment:

_Fill out and check (`[x]`) the boxes which apply. If your Fail2Ban version is outdated, 
and you can't verify that the issue persists in the recent release, better seek support 
from the distribution you obtained Fail2Ban from_

- Fail2Ban version (including any possible distribution suffixes): 0.11.1-r2
- OS, including release name/version: Gentoo unstable
- [x] Fail2Ban installed via OS/distribution mechanisms
- [x] You have not applied any additional foreign patches to the codebase
- [x] Some customizations were done to the configuration (provide details below is so)

### The issue:

syslog fmt has unwanted space between process name and [pid], breaking many standard log parsers.

#### Steps to reproduce

Run fail2ban with `logtarget = SYSLOG`, get some ban events.

#### Expected behavior

Syslog entries like:

`May 25 15:15:00 foo fail2ban.actions[12345]: NOTICE [jailname] Ban 1.2.3.4`

#### Observed behavior

`May 25 15:14:00 foo fail2ban.actions [12121]: NOTICE [jailname] Ban 1.2.3.4`

#### Any additional information

This is a continuation or re-emergence of issue #2099

I've sprinkled debug prints to see what is going on. `padding` is defaulting to off as intended, but the replacement when fmt is rewritten is not consuming the space between the name and process ID.

This patch seems to do what is intended:

```diff
# diff -u fail2ban/helpers.py.orig fail2ban/helpers.py
--- fail2ban/helpers.py.orig    2020-05-24 17:23:26.000000000 -0400
+++ fail2ban/helpers.py 2020-05-25 15:26:42.746042886 -0400
@@ -273,7 +273,7 @@
                        fmt = "%(asctime)s " + fmt
        # remove padding if not needed:
        if not padding:
-               fmt = re.sub(r'(?<=\))-?\d+(?:\.\d+)?s', lambda m: 's', fmt)
+               fmt = re.sub(r'(?<=\))-?\d+(?:\.\d+)?s ?', lambda m: 's', fmt)
        return fmt
 
```

In the meantime I am using a similar workaround to what is mentioned in #2099, in fail2ban.conf:

```ini
logtarget = SYSLOG[format="%%(name)s[%%(process)d]: %%(levelname)s %%(message)s"]
```

But that is not an ideal long-term solution.

#### Any customizations done to /etc/fail2ban/ configuration
Many, but AFAIK nothing relevant.

#### Relevant parts of /var/log/fail2ban.log file:
_preferably obtained while running fail2ban with `loglevel = 4`_

```
```

#### Relevant lines from monitored log files in question:

```
```

**Comment by sebres** [comment]
Confirmed... Thx!

> This patch seems to do what is intended:
```diff
-               fmt = re.sub(r'(?<=\))-?\d+(?:\.\d+)?s', lambda m: 's', fmt)
+               fmt = re.sub(r'(?<=\))-?\d+(?:\.\d+)?s ?', lambda m: 's', fmt)
```

Hmmm... I don't think this is correct since this will remove every single space after `%(xxx)-Ns` also if it is mandatory space.
I think the extra padding space that you wanted not to see now, is a space between `fail2ban.actions` and `[PID]` only, all other (single) spaces are rather mandatory and should remain.

#### PoC:
```python
>>> import re; fmt = '%(xxx)-10s [%(pid)s] %(2nd)-7s %(3rd)s'; 
>>> print(fmt); print(re.sub(r'(?<=\))-?\d+(?:\.\d+)?s ?', lambda m: 's', fmt))
%(xxx)-10s [%(pid)s] %(2nd)-7s %(3rd)s
%(xxx)s[%(pid)s] %(2nd)s%(3rd)s
```
As you can see the space between 2nd and 3rd parameters are away too. And since it is not really a padding, this is incorrect.

Sure one could extend the padding removal RE a bit more precise like:
```diff
-re.sub(r'(?<=\))-?\d+(?:\.\d+)?s', lambda m: 's', fmt)
+re.sub(r'(?<=\))-?\d+(?:\.\d+)?s(?: (?=\[))?', lambda m: 's', fmt)
```
But I guess removing space before `[` in a given format would be more practicable.

**Comment by sebres** [comment]
Should be fixed in 9b6da03c90139db1a8820a24736462c5c3477ba8 now.

**Comment by hlein** [comment]
Hah, I think you are right. Thanks for the quick better fix!
